usb/usb.c

```diff
@@ -100,14 +108,20 @@ void usb_transferFinished(usb_transfer_t *t, int status)
 		t->error = -status;
 	}
 
-	/* URB Transfer */
-	if (t->msg != NULL)
+	/* Internal transfer */
+	if (t->port == 0) {
+		if (t->type == usb_transfer_interrupt && t->transferred > 0)
+			hub_notify(t->hub);
+		else
+			usb_devSignal();
+	}
+	else {
+		/* URB transfer */
+		t->state = urb_completed;
 		LIST_ADD(&usb_common.finished, t);
-	else if (t->type == usb_transfer_interrupt && t->transferred > 0)
-		hub_notify(t->pipe->dev);
+		condSignal(usb_common.finishedCond);
```

Comment:
Is this cond guarded by transferLock? If so it would be better to move this signal outside somehow (a flag?)